src/main/kotlin/ar/edu/unahur/obj2/socios/Barrios.kt

```diff
@@ -0,0 +1,26 @@
+package ar.edu.unahur.obj2.socios
+
+abstract class Barrio{
+    abstract fun cuenta(monto: Double): Double
+}
+
+object lasRosas : Barrio() { //agrega un plus fijo de $50 a la propina,
+    override fun cuenta(monto: Double) = monto + 50
+}
+
+object lasLauchas : Barrio() { //reduce a la mitad la propina,
+    override fun cuenta(monto: Double) = monto / 2
+}
+
+object barrioVerde : Barrio() { //fija un mínimo de $200 para la propina,
+    override fun cuenta(monto: Double): Double {
+        return when {
+            monto <= 200 -> 200.0
+            else -> monto
+        }
+    }
+}
```

Comment:
Acá podrían haber usado `max()`. Si el valor es menor a 200 devuelve 200 y si es mayor devuelve el monto.
```suggestion
    override fun cuenta(monto: Double) = max(monto, 200.0)
```